doc/autoscale/experiment/python/collector.py

```diff
@@ -0,0 +1,131 @@
+from pprint import pprint
+import time
+
+JOB_STATUS_NOT_EXISTS = 0
+JOB_STATUS_PENDING = 1
+JOB_STATUS_RUNNING = 2
+JOB_STATUS_FINISHED = 3
+
+class JobInfo(object):
+    def __init__(self, name):
+        self.name = name
+        self.started = False
+        self.status = JOB_STATUS_NOT_EXISTS
+        self.submit_time = 0
+        self.start_time = 0
+        self.end_time = 0
+
+    def status_str(self):
+        if self.status == JOB_STATUS_FINISHED:
+            return 'FINISHED'
+        elif self.status == JOB_STATUS_PENDING:
+            return 'PENDING'
+        elif self.status == JOB_STATUS_NOT_EXISTS:
+            return 'NOT_EXISTS'
+        elif self.status == JOB_STATUS_RUNNING:
+            return 'RUNNING'
+
+
+class Collector(object):
+    '''
+    Collector monitor data from Kubernetes API
+    '''
+    def __init__(self):
+        # TODO(Yancey1989): 
+        #   init kubernetes configuration 
+        #   from settings.py
+        config.load_kube_config()
+        self.cpu_allocatable = 0
+        self.gpu_allocatable = 0
+        self.cpu_requests = 0
+        self.gpu_requests = 0
+
+        # Collect cluster wide resource
+        self._init_allocatable()
+
+        self._pods = []
+
+    def _init_allocatable(self):
+        api_instance = client.CoreV1Api()
+        try: 
+            api_response = api_instance.list_node()
+            cpu = 0
+            gpu = 0
+            for item in api_response.items:
+                allocate = item.status.allocatable
+                cpu += int(allocate.get('cpu', 0))
+                gpu += int(allocate.get('gpu', 0))
+            self.cpu_allocatable = cpu
+            self.gpu_allocatable = gpu
+        except ApiException as e:
+            print("Exception when calling CoreV1Api->list_node: %s\n" % e)
+    
+    def _real_cpu(self, cpu):
+        if cpu:
+            if cpu.endswith('m'):
+                return 0.001 * int(cpu[:-1])
+            else:
+                return int(cpu)
+        return 0
+    
+
+    def run_once(self):
+        api_instance = client.CoreV1Api()
+        try:
+            api_response = api_instance.list_pod_for_all_namespaces(pretty=True)
+            self._pods = api_response.items
+        except ApiException as e:
+            print("Exception when calling CoreV1Api->list_pod_for_all_namespaces: %s\n" % e)
+        return int(time.time())
+
+    def cpu_utils(self):
+        cpu = 0
+        for item in self._pods:
+            for container in item.spec.containers:
+                requests = container.resources.requests
+                if requests:
+                    cpu += self._real_cpu(requests.get('cpu', None))
+        
+        return '%0.2f' % ((100.0 * cpu) / self.cpu_allocatable)
+
+    def gpu_utils(self):
+        gpu = 0
+        for item in self._pods:
+            for container in item.spec.containers:
+                requests = container.resources.requests
+                if requests:
```

Comment:
the documentation seems to be using `limits` rather than `requests` for GPU (feels a little strange): https://kubernetes.io/docs/tasks/manage-gpus/scheduling-gpus/#api